examples/query_expansions/run.ts

```diff
@@ -0,0 +1,92 @@
+import Instructor from "@/instructor"
+import readline from 'readline';
+import { OpenAI } from "openai";
+
+import { z } from "zod"
+
+
+const getSystem = () => {
+  const currentDate = new Date();
+  const formattedDate = currentDate.toLocaleDateString();
+  const time = currentDate.toLocaleTimeString();
+  const systemPrompt = `
+  You are a world class query understanding algorithm that is able to:
+
+  1. rewrite queries to be specific to the context, include additional terms that are relevant to the context, and remove terms that are not relevant to the context.
+  2. consider the date and relative and absolute time of the query and the context.
+  3. When the query is the search query should be fully qualified, and contain all themes, entities, and keywords that are relevant to the context.
+
+  Tips:
+    * PERSONAL_DATA: The query is asking for personal data. The answer should be a list of personal data.
+    * INTERNET: The query is asking for information on the internet or latest information that may not be available in the context.
+    * TRANSCRIPTS: The query is asking for a transcript of a conversation or a meeting.
+
+  The current date is ${formattedDate} and the current time is ${time}.
+  `;
+
+  return systemPrompt;
+};
+
+
+const ExtractionValuesSchema = z.object({
+  rewrittenQuery: z.string().describe("Rewrite the query to be specific to the context. This will be used to do semantic search, so make sure it is specific to the context."),
+  questionType: z.array(z.string().describe("The type of question that is being asked. This will be used to determine the type of answer that is expected. MUST be one of the following: PERSONAL_DATA, INTERNET, TRANSCRIPTS")),
```

Comment:
could use a .enum(["PERSONAL_DATA", ...]) here?

Reply:
enum does not work while streaming

Reply:
hmm - why not? its just being set on the json schema right? Feel like I have had better accuracy defining enums - although I have 0 actual data to back up that claim lol

https://ss.dimitri.link/0G9V4s5n